### ISSUE TITLE
Scrub credentials in bind requests

### ISSUE THREAD

**Issue by michaelklishin** [rabbitmq/rabbitmq-auth-backend-ldap#12]
When a user configures logging level to trace LDAP traffic, we log credentials in bind requests. While I'm not sure we can scrub all credentials — LDAP objects can contain any data, after all — we likely can do that for bind requests.

**Comment by Ayanda-D** [comment]
@michaelklishin / @hairyhum I just need some clarity on this feature. Consider the following log extracts;
- Login (user) credential and bind dn credential different. Makes sense to scrub credential from bind dn only.

```
=INFO REPORT==== 22-Apr-2016::10:55:59 ===
LDAP CHECK: login for Alice

=INFO REPORT==== 22-Apr-2016::10:55:59 ===
    LDAP bind succeeded: cn=admin,dc=example,dc=com
```
- Login credential/username and credential in bind DN the same. Should we also scrub the credential from the login info report?

```
=INFO REPORT==== 22-Apr-2016::10:56:00 ===
LDAP CHECK: login for Alice

=INFO REPORT==== 22-Apr-2016::10:56:00 ===
        LDAP filling template "cn=${username},ou=People,dc=example,dc=com" with
            [{username,<<"Alice">>}]

=INFO REPORT==== 22-Apr-2016::10:56:00 ===
        LDAP template result: "cn=Alice,ou=People,dc=example,dc=com"

=INFO REPORT==== 22-Apr-2016::10:56:00 ===
    LDAP bind succeeded: cn=Alice,ou=People,dc=example,dc=com
```

The latter could mean introducing a global configuration variable to turn scrubbing of (all) credentials in the logs on/off. But the value of having this is questionable, but would be at the user's will.

Then an example of a scrubbed DN, (or do we want to scrub more elements besides **cn** and **uid**?); 
**cn=Alice,ou=People,dc=example,dc=com** would be **cn=xxxxx,ou=People,dc=example,dc=com**

**Comment by Ayanda-D** [comment]
Correction (or addition) to the question above. **network** logging configured;

```

=INFO REPORT==== 22-Apr-2016::13:35:27 ===
    LDAP network traffic: bind request = {'BindRequest',3,
                                          "cn=Alice,ou=People,dc=example,dc=com",
                                          {simple,<<"password">>}}

=INFO REPORT==== 22-Apr-2016::13:35:27 ===
    LDAP network traffic: bind reply = {ok,
                                        {'LDAPMessage',69,
                                         {bindResponse,
                                          {'BindResponse',success,[],[],
                                           asn1_NOVALUE,asn1_NOVALUE}},
                                         asn1_NOVALUE}}
```

So scrubbing password, definitely, question is still on the DN? How far do we take this? Are we scrubbing entire DN, or just some of its elements?

**Comment by hairyhum** [comment]
Fixed by #31
